### PR TITLE
Consolidating the Solution Interface

### PR DESCRIPTION
**Pull request by echandler5956f**
Addressed Issue #65 (retrieving the solution is now ~500x faster) and updated some much-needed documentation.

### CODE REVIEW

File: galileo/opt/include/galileo/opt/TrajectoryOpt.h [hunk 1]
```diff
@@ -331,12 +392,12 @@ namespace galileo
...
             std::shared_ptr<DecisionData> Wdata = std::make_shared<DecisionData>();
 
-            auto num_phases = sequence->getNumPhases();
+            size_t num_phases = sequence->getNumPhases();
 
             printf("Starting initialization\n");
             auto begin_time = std::chrono::high_resolution_clock::now();
             auto start_time = std::chrono::high_resolution_clock::now();
-            auto end_time = std::chrono::high_resolution_clock::now();  
+            auto end_time = std::chrono::high_resolution_clock::now();
```
Comment (Akshay5312) [inline galileo/opt/include/galileo/opt/TrajectoryOpt.h:400]: I would remove the timing, and the dependency on chrono

It's not needed in  the actual evaluation. 

Possibly add an "enable timing" option, if this is really needed for testing/benchmarking

File: galileo/opt/include/galileo/opt/TrajectoryOpt.h [hunk 2]
```diff
@@ -555,86 +616,55 @@ namespace galileo
         }
 
         template <class ProblemData, class MODE_T>
-        void TrajectoryOpt<ProblemData, MODE_T>::getSolution(solution_t &result)
+        void TrajectoryOpt<ProblemData, MODE_T>::getSolutionSegments(std::vector<solution::solution_segment_data_t> &result, Eigen::VectorXd &times)
```
Comment (Akshay5312) [inline galileo/opt/include/galileo/opt/TrajectoryOpt.h:619]: yeah, it doesn't look like you modify times. Const may be best

File: galileo/opt/src/Solution.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                            double input_scaled_time = (query_times(i) - input_knot_start_time) / (input_knot_end_time - input_knot_start_time);
+                            input_result.col(i) = solution_segments_[j].input_poly.barycentricInterpolation(input_scaled_time, input_terms);
+                            break;
+                        }
+                    }
+                }
+
+                auto clock_end_time = std::chrono::high_resolution_clock::now();
+                std::chrono::duration<double> elapsed_time = clock_end_time - clock_start_time;
+                std::cout << "GetSolution took " << elapsed_time.count() << " seconds" << std::endl;
```
Comment (Akshay5312) [inline galileo/opt/src/Solution.cpp:49]: same comment, getting this data probably won't be necessary to a user, right?

**Review by Akshay5312 — COMMENTED** [review]
Looks great! 

I think maybe there's two style things, that's it

File: galileo/opt/include/galileo/opt/TrajectoryOpt.h (hunk 2 above)
Comment (echandler5956f) [reply galileo/opt/include/galileo/opt/TrajectoryOpt.h]: Fixed!

File: galileo/opt/src/Solution.cpp (hunk 3 above)
Comment (echandler5956f) [reply galileo/opt/src/Solution.cpp:49]: We will remove this later down the line. The timing info is useful for debugging and development right now.

**Pull request merged**